gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+class ApplicationSeparator(metaclass=ABCMeta):
+    @abstractmethod
+    def is_supported(self, process: Process) -> bool:
+        pass
+
+    @abstractmethod
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+    @property
+    @abstractmethod
+    def app_prefix(self) -> str:
+        pass
+
+
+IP_PORT_RE = re.compile(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(?:\:\d{2,5})?$")  # Matches against (ip(:port))
+
+
+def get_positional_arg(args: List[str], pos_arg_name: str) -> Optional[str]:
+    if pos_arg_name in args:
+        return args[args.index(pos_arg_name) + 1]
+
+    return None
+
+
+def append_python_module_to_proc_wd(process: Process, module_name: str) -> str:
+    return process.cwd().replace("/", ".") + module_name
```

Comment:
`process.cwd()` returns an absolute path, so for a module named "my-module", if I ran it from `/home/myself` I would get `'.home.myselfmy-module'`. It doesn't seem right...

Reply:
It's correct that it returns an absolute path, it had a bug that's why it wasn't seemed right, now I fixed it and the result of the function will be: `home.myself.my-module` which is what I wanted.
The intention of the function is to add the workdir since many applications run as as module `wsgi/app`  which is not indicative at all, but when adding the working directory it does add some indicative data.

Reply:
It's a good point. But I think that `home.myself.my-module` is not the correct way to display the path because it looks like the name of the module, which it isn't so it looks like a bug. I think we should display the module name and the path separately like in Python frames, i.e. `module_name (file_path`).

As a separated point, it's worth checking if the regexes that collapses generated paths to i.e. `\m\` when displayed in the UI work with these changes.

Reply:
Agreed, fixed

Reply:
I agree with @YishaiZinkin 's suggestion.

@guybortnikov , I don't see it pushed..

Reply:
Done, resolving